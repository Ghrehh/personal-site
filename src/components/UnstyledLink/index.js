// @flow
import * as React from 'react';
import { Link } from 'react-router-dom';

import styles from './styles.module.css';

type Props = {
  children: React.Node,
  link: string,
  external: boolean
};

const LinkButton = (props: Props) => {
  if (props.external) {
    return (
      <a target="_blank" href={props.link} className={styles.link}>
        {props.children}
      </a>
    );
  }

  return (
    <Link to={props.link} className={styles.link}>
      {props.children}
    </Link>
  );
};

LinkButton.defaultProps = {
  external: false
};

export default LinkButton;
