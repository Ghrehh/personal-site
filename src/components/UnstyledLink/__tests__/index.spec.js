import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import UnstyledLink from '../';

jest.mock('react-router-dom', () => ({ Link: 'div' }));

describe('UnstyledLink', () => {
  let testContext = {};

  beforeEach(() => {
    testContext = {};

    testContext.component = <UnstyledLink link="./page">Foo</UnstyledLink>;
  });

  describe('rendering', () => {
    beforeEach(() => {
      testContext.renderer = new ShallowRenderer();
      testContext.renderer.render(testContext.component);
      testContext.result = testContext.renderer.getRenderOutput();
    });

    it('renders the component correctly', () => {
      expect(testContext.result).toMatchSnapshot();
    });
  });
});
