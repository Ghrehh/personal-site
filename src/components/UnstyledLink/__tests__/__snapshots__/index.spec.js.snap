// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnstyledLink rendering renders the component correctly 1`] = `
<div
  className="link"
  to="./page"
>
  Foo
</div>
`;
